source/base/polynomials_bernstein.cc

```diff
@@ -38,11 +17,13 @@ namespace Binomial
     AssertIndexRange(k, n+1);
     std::vector<number> coeff(n + 1, number(0.0));
     for (unsigned int i = k; i < n + 1; ++i)
-      coeff[i] = (pow(number(-1), number(i - k)) * binomial(n, i)
-                  * binomial(i, k));
+      {
+        coeff[i] = ((i - k) % 2 == 0 ? 1 : -1)
+                   * boost::math::binomial_coefficient<number>(n, i)
+                   * boost::math::binomial_coefficient<number>(i, k);
```

Comment:
For the record, I just compiled everything with the bundled boost and it continues to compile fine.